src/main/java/org/apache/bcel/generic/ReferenceType.java

```diff
@@ -222,9 +222,10 @@ public boolean isAssignmentCompatibleWith( final Type t ) throws ClassNotFoundEx
             if ((arrType1.getDimensions() == arrType2.getDimensions())
                     && arrType1.getBasicType() instanceof ObjectType
                     && arrType2.getBasicType() instanceof ObjectType) {
-                return new ArrayType(((ObjectType) arrType1.getBasicType())
-                        .getFirstCommonSuperclass((ObjectType) arrType2.getBasicType()), arrType1
-                        .getDimensions());
+                ObjectType basicType1 = (ObjectType) arrType1.getBasicType();
```

Comment:
This change is probably better style, but it is a gratuitous difference from upstream.  Please undo it.